java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -141,6 +155,8 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
     out.align();
 
     long bufferLength = writeBatchBuffers(out, batch);
+    // align again - the size is captured in out bodyLength check above
+    bufferLength += out.align();
```

Comment:
Per above, `bufferLength` here should already be a multiple of 8, and it may be worth asserting here (e.g. in C++ we have debug assertions already about this). The padding should be handled in `writeBatchBuffers`, and it seems potentially dangerous to handle it anywhere else